plugins/inputs/fibaro/fibaro_test.go

```diff
@@ -205,3 +274,69 @@ func TestJSONSuccess(t *testing.T) {
 	fields = map[string]interface{}{"energy": float64(4.33), "power": float64(0.7), "value": float64(50), "value2": float64(75)}
 	acc.AssertContainsTaggedFields(t, "fibaro", fields, tags)
 }
+
+func TestHC3JSON(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		payload := ""
+		switch r.URL.Path {
+		case "/api/sections":
+			payload = sectionsJSON
+		case "/api/rooms":
+			payload = roomsJSON
+		case "/api/devices":
+			payload = devicesHC3JSON
+		}
+		w.WriteHeader(http.StatusOK)
+		_, err := fmt.Fprintln(w, payload)
+		require.NoError(t, err)
+	}))
+	defer ts.Close()
+
+	a := Fibaro{
+		URL:        ts.URL,
+		Username:   "user",
+		Password:   "pass",
+		DeviceType: "HC3",
+		client:     &http.Client{},
+	}
+	require.NoError(t, a.Init())
+
+	var acc testutil.Accumulator
+	err := acc.GatherError(a.Gather)
+	require.NoError(t, err)
+
+	// Gather should add 5 metrics
+	require.Equal(t, uint64(5), acc.NMetrics())
+
+	// Ensure fields / values are correct - Device 1
+	tags := map[string]string{"deviceId": "1", "section": "Section 1", "room": "Room 1", "name": "Device 1", "type": "com.fibaro.binarySwitch"}
+	fields := map[string]interface{}{"value": float64(0)}
+	acc.AssertContainsTaggedFields(t, "fibaro", fields, tags)
+
+	// Ensure fields / values are correct - Device 2
+	tags = map[string]string{"deviceId": "2", "section": "Section 2", "room": "Room 2", "name": "Device 2", "type": "com.fibaro.binarySwitch"}
+	fields = map[string]interface{}{"value": float64(1)}
+	acc.AssertContainsTaggedFields(t, "fibaro", fields, tags)
+
+	// Ensure fields / values are correct - Device 3
+	tags = map[string]string{"deviceId": "3", "section": "Section 3", "room": "Room 3", "name": "Device 3", "type": "com.fibaro.multilevelSwitch"}
+	fields = map[string]interface{}{"value": float64(67)}
+	acc.AssertContainsTaggedFields(t, "fibaro", fields, tags)
+
+	// Ensure fields / values are correct - Device 4
+	tags = map[string]string{"deviceId": "4", "section": "Section 3", "room": "Room 4", "name": "Device 4", "type": "com.fibaro.temperatureSensor"}
+	fields = map[string]interface{}{"batteryLevel": float64(100), "value": float64(22.8)}
+	acc.AssertContainsTaggedFields(t, "fibaro", fields, tags)
+
+	// Ensure fields / values are correct - Device 5
+	tags = map[string]string{"deviceId": "5", "section": "Section 3", "room": "Room 4", "name": "Device 5", "type": "com.fibaro.FGRM222"}
+	fields = map[string]interface{}{"energy": float64(4.33), "power": float64(0.7), "value": float64(34)}
+	acc.AssertContainsTaggedFields(t, "fibaro", fields, tags)
```

Comment:
Not sure, but I would prefer defining the expected metrics and use `testutil.RequireMetricsEqual`...